### PR TITLE
p3_path_link test

### PR DESCRIPTION
**Pull request by squillace**
a tentative test case for wasmtime p3_path_link by automating the conversion of p1 tests to p3. The first few PRs like this one are to take care to review the specific case to determine whether the automation is in fact doing the right thing. -- ralph

### CODE REVIEW

File: crates/wasi/tests/all/p3/mod.rs [hunk 1]
```diff
@@ -164,3 +164,13 @@ async fn p3_file_write() -> wasmtime::Result<()> {
...
+
+#[test_log::test(tokio::test(flavor = "multi_thread"))]
+async fn p3_path_link() -> wasmtime::Result<()> {
+    run(P3_PATH_LINK_COMPONENT).await
+}
+
+#[test_log::test(tokio::test(flavor = "multi_thread"))]
+async fn p3_path_link_blocking() -> wasmtime::Result<()> {
+    run_allow_blocking_current_thread(P3_PATH_LINK_COMPONENT, true).await
+}
```
Comment (alexcrichton) [inline crates/wasi/tests/all/p3/mod.rs:176]: I think that this is fine to omit, most of the tests here don't have this variant and IIRC for the above two (or maybe just readdir?) it was a specific edge case that was being tested. Or at least I think, maybe I'm misremembering...

**Review by alexcrichton — APPROVED** [review]
Thanks! Test looks to me to be a faithful translation (mostly). The missing piece here is the `config().support_dangling_filesystem()`-guarded part of the original test, which is a bit esoteric. I believe though that the same parts of that test can be included here too?

One thing that might also be nice is to separate out each of these individual tests to their own functions to make it a bit clearer that there's a bunch of test cases. It was all glommed together in the p1 test but I think it'd be fine to refactor-as-we-go into p3.

**Comment by alexcrichton** [comment]
Also, of note, the CI failure with rustfmt should be easy enough to fix

File: crates/wasi/tests/all/p3/mod.rs (hunk 1 above)
Comment (squillace) [reply crates/wasi/tests/all/p3/mod.rs:176]: I'll have a deeper look. Thanks for thinking about this.
